pkg/resource/deploy/step.go

```diff
@@ -337,26 +337,34 @@ func (s *DeleteStep) Logical() bool           { return !s.replacing }
 func (s *DeleteStep) Apply(preview bool) (resource.Status, StepCompleteFunc, error) {
 	// Refuse to delete protected resources (unless we're replacing them in
 	// which case we will of checked protect elsewhere)
-	if !s.replacing && s.old.Protect {
+	if !s.replacing && (s.old.Protect || s.old.DeleteBehaviour == resource.DeleteBehaviourProtect) {
```

Comment:
Repeating a qn here - Protect and DeleteBehaviorProtect seem awfully similar, are they different somehow that we want both?